src/Microsoft.Framework.Logging/Logger.cs

```diff
@@ -29,9 +30,28 @@ public void Log(LogLevel logLevel, int eventId, object state, Exception exceptio
         {
             if (logLevel >= _loggerFactory.MinimumLevel)
             {
+                List<Exception> exceptions = null;
                 foreach (var logger in _loggers)
                 {
-                    logger.Log(logLevel, eventId, state, exception, formatter);
+                    try
+                    {
+                        logger.Log(logLevel, eventId, state, exception, formatter);
+                    }
+                    catch (Exception ex)
+                    {
+                        if (exceptions == null)
+                        {
+                            exceptions = new List<Exception>(_loggers.Length);
```

Comment:
Why initialize the length of the list? That assumes that *all* loggers will throw, and pre-allocates the list to have that many slots. I would think the common case is that if a logger fails, it's just that one failure.